src/main/java/teammates/ui/webapi/action/GetCourseAction.java

```diff
@@ -16,20 +19,42 @@ protected AuthType getMinAuthLevel() {
 
     @Override
     public void checkSpecificAccessControl() {
+        if (!(userInfo.isInstructor || userInfo.isStudent)) {
+            throw new UnauthorizedAccessException("Student or instructor account is required to access this resource.");
+        }
+
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
 
-        // TODO enable access for student
-        gateKeeper.verifyAccessible(
-                logic.getInstructorForGoogleId(courseId, userInfo.getId()),
-                logic.getCourse(courseId));
+        if (userInfo.isInstructor) {
+            gateKeeper.verifyAccessible(
+                    logic.getInstructorForGoogleId(courseId, userInfo.getId()),
+                    logic.getCourse(courseId));
+        }
+
+        if (userInfo.isStudent) {
+            if (!hasStudentJoinedCourse(courseId)) {
```

Comment:
This `if` check is not necessary I think. `gateKeeper.verifyAccessible(logic.getStudentForGoogleId(courseId, userInfo.id), course);` will do null check.